File: ManejadorReglas.py
class ManejadorReglas:

    def __init__(self):
        self.__regla=None

    def set_regla(self, regla):
        self.__regla = regla
    
    def ejecutar_regla(self, cantidad, precio):
        return self.__regla.calcular_total(cantidad, precio)

File: Backend/model/ManejadorReglas.ts
import ReglaEspecial from "./ReglaEspecial";
import ReglaNormal from "./ReglaNormal";
import ReglaPeso from "./ReglaPeso";
import ReglaPrecio from "./ReglaPrecio";

export default class ManejadorReglas{
    private regla: ReglaPrecio[];

    constructor(){
        this.regla = [new ReglaNormal(), new ReglaPeso(), new ReglaEspecial()];
    }

    public ejecutarRegla(cantidad:number, precio:number, sku:string): number | undefined{
        return this.regla.filter(regla => regla.aplica(sku) === true)[0].calcularTotal(cantidad, precio)
    }
}